mitodl/mitxpro
# Add support for Global Data Consent Agreement

#### Pre-Flight checklist




#### What are the relevant tickets?
#2178 

#### What's this PR do?
- Adds support for Global Consent Agreements

#### How should this be manually tested?
It might be a good idea to read the comments on issue #2178 and then:
1) Create a company (Optional)
2) Create coupon for a specific company using `ecommerce/admin/coupons/` and download the coupon codes
3) Create a consent agreement for a specific company and select the `All Courses` checkbox
4) Now apply a code from the csv downloaded in step 2
5) You should see a `Consent Agreement Message` at the bottom of the checkout page and upon clicking `Data Sharing Policy` You should see the `Content` added while creating the agreement
6) Now create another Consent Agreement for the same company and keep the `All Courses` checkbox unchecked and add a specific course to the list
7) Follow the steps for checkout and this time upon clicking `Data Sharing Policy` you should see course-based consent for the company.
8) The above scenario for Global consent should work for a course that is newly created and isn't added in any `Consent Agreement` here (`admin/ecommerce/dataconsentagreement`)


#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
You can follow the associated ticket for more context

#### Screenshots (if appropriate)
**Validation message for Either making the agreement global or associate it with course**

<img width="1391" alt="Screenshot 2021-05-05 at 5 22 30 PM" src="https://user-images.githubusercontent.com/34372316/117141253-c202e980-adc7-11eb-82f0-31ce728a052c.png">

**Validation message for an existing Global agreement for a company since we expect only one global agreement per company**
<img width="974" alt="Screenshot 2021-05-05 at 5 24 49 PM" src="https://user-images.githubusercontent.com/34372316/117141271-c62f0700-adc7-11eb-8187-f92ffe8b0efd.png">

**Checkout Page with Consent checkbox**
<img width="1251" alt="Screenshot 2021-05-05 at 5 25 55 PM" src="https://user-images.githubusercontent.com/34372316/117141272-c6c79d80-adc7-11eb-92ef-f8dc42f6459e.png">

**Checkout page with Course based consent (Case when there is a course based consent available)**
<img width="992" alt="Screenshot 2021-05-05 at 5 26 08 PM" src="https://user-images.githubusercontent.com/34372316/117141281-c7f8ca80-adc7-11eb-9f75-29a4cf809cc6.png">

**Checkout page with Global Consent (Case when there is no course based consent available)**

<img width="1262" alt="Screenshot 2021-05-05 at 5 27 11 PM" src="https://user-images.githubusercontent.com/34372316/117141286-c8916100-adc7-11eb-870d-5458ffda40ca.png">
